python/lsst/drp/tasks/update_visit_summary.py

```diff
@@ -0,0 +1,755 @@
+            References to the catalog datasets to load.
+
+        Returns
+        -------
+        wrapper : `PerTractInput`
+            Wrapper object for the loaded catalogs.
+        """
+        catalogs_by_tract = []
+        for ref in refs:
+            tract_id = ref.dataId["tract"]
+            tract_info = sky_map[tract_id]
+            catalogs_by_tract.append(
+                (
+                    tract_info,
+                    butler.get(ref),
+                )
+            )
+        return cls(catalogs_by_tract)
+
+    def best_for_detector(
+        self,
+        detector_id: int,
+        center: SpherePoint | None = None,
+        bbox: Box2I | None = None,
+    ) -> tuple[int, ExposureRecord | None]:
+        # Docstring inherited.
+        best: tuple[int, ExposureRecord | None] = (-1, None)
+        best_distance: Angle = float("inf") * degrees
+        for tract_info, catalog in self.catalogs_by_tract:
+            record = catalog.find(detector_id)
+            if record is None:
+                continue
+            if center is None:
+                center_for_record = compute_center_for_detector_record(
+                    record, bbox=bbox
+                )
+                if center_for_record is None:
+                    continue
+            else:
+                center_for_record = center
+            center_distance = tract_info.ctr_coord.separation(center_for_record)
+            if best_distance > center_distance:
+                best = (tract_info.tract_id, record)
+                best_distance = center_distance
+        return best
+
+
+@dataclasses.dataclass
+class GlobalInput(PossiblyMultipleInput):
+    """Wrapper class for input `~lsst.afw.table.ExposureCatalog` datasets
+    that are not per-tract.
+    """
+
+    catalog: ExposureCatalog
+    """Loaded per-visit catalog dataset (`lsst.afw.table.ExposureCatalog`).
+    """
+
+    def best_for_detector(
+        self,
+        detector_id: int,
+        center: SpherePoint | None = None,
+        bbox: Box2I | None = None,
+    ) -> tuple[int, ExposureRecord | None]:
+        # Docstring inherited.
+        return -1, self.catalog.find(detector_id)
+
+
+class UpdateVisitSummaryConnections(
+    PipelineTaskConnections,
+    dimensions=("instrument", "visit"),
+    defaultTemplates={
+        "skyWcsName": "jointcal",
+        "photoCalibName": "fgcm",
+    },
+):
+    input_summary_schema = cT.InitInput(
+        doc="Schema for input_summary_catalog.",
+        name="visitSummary_schema",
+        storageClass="ExposureCatalog",
+    )
+    sky_map = cT.Input(
+        doc="Description of tract/patch geometry",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+    input_summary_catalog = cT.Input(
+        doc="Visit summary table to load and modify.",
+        name="visitSummary",
+        dimensions=(
+            "instrument",
+            "visit",
+        ),
```

Comment:
No, not necessary.  In case you're curious, it happened here becase black takes a trailing comma as a hint to format like this instead of on one line.